devtools: enables typescript `strict` option

JeanMeche opened the pull request:
Enabling `strict` is part of an effort to improve the quality of the devtools code base. 
One of the direct side effect is to enable `noImplicitAny`, `strictPropertyInitialization` and `strictBindCallApply`.


This commit also replaces `fullTemplateTypeCheck` with `stringTemplates`.

cc @ducin

kbrilla commented on devtools/projects/ng-devtools-backend/src/lib/directive-forest/ltree.ts:
```diff
@@ -37,7 +41,7 @@ const isLView = (value: any): boolean => {
 
 export const METADATA_PROPERTY_NAME = '__ngContext__';
-export const getLViewFromDirectiveOrElementInstance = (dir: any): null|{} => {
+export function getLViewFromDirectiveOrElementInstance(dir: any): null|LView {
```
any reason this is converted to function while on line 35 `isLContainer`  is kept as const?

JeanMeche commented on devtools/projects/ng-devtools-backend/src/lib/state-serializer/state-serializer.ts:
```diff
@@ -51,38 +51,39 @@ const nestedSerializerContinuation = (nodes: NestedProp[], level: number) =>
+    if (typeof prop === 'string' && ignoreList.has(prop)) {
+      return;
+    }
+    // TODO: investigate why this call didn't match the signature
```
This one is particularly suspicious.

JeanMeche committed:
refactor(devtools): enables typescript strict option

Enabling `strict` is part of an effort to improve the quality of the devtools code base.
One of the direct side effect is to enable `noImplicitAny`, `strictPropertyInitialization` and `strictBindCallApply`.

This commit also replaces `fullTemplateTypeCheck` with `stringTemplates`.

AleksanderBodurri reviewed (approved):
Awesome stuff! Thanks @JeanMeche

dylhunn commented:
We discussed, and Alex's approval should be sufficient

dylhunn commented:
This PR was merged into the repository by commit 6cd91c675c8d248736cfe447f2a5d6dc488093b8.
